cli/pcluster/config/cfn_param_types.py

```diff
@@ -816,6 +816,38 @@ def refresh(self):
             self.owner_section.get_param("architecture").value = architecture
 
 
+class ArgsCfnParam(CfnParam):
+    """
+    Class to manage the pre/post_install_args configuration parameter.
+
+    We need this class in order to escape the args to json param.
+    """
+
+    def from_cfn_params(self, cfn_params):
+        """
+        Initialize parameter value by parsing CFN input parameters.
+
+        :param cfn_params: list of all the CFN parameters, used if "cfn_param_mapping" is specified in the definition
+        """
+        cfn_converter = self.definition.get("cfn_param_mapping", None)
+        if cfn_params:
+            cfn_value = get_cfn_param(cfn_params, cfn_converter) if cfn_converter else "NONE"
+            self.value = self.get_value_from_string(json.loads('"' + cfn_value + '"'))
+
+        return self
+
+    def to_cfn(self):
+        """Convert param to CFN representation, if "cfn_param_mapping" attribute is present in the Param definition."""
+        cfn_params = {}
+        cfn_converter = self.definition.get("cfn_param_mapping", None)
+
+        if cfn_converter:
+            cfn_value = self.get_cfn_value()
+            cfn_params[cfn_converter] = json.dumps(cfn_value)[1:-1]
```

Comment:
I assume this is done to remove `""` from `cfn_value`, is there a way to do this not with indexing? I am just afraid we might remove something unexpected silently. If not, please add a comment in the code here clarifying why this `[1:-1]` logic is needed